framework/src/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -3106,7 +3106,7 @@ public boolean validateTypeOf(Tree tree) {
             }
             break;
         default:
-            type = atypeFactory.getAnnotatedType(tree);
+        	type = atypeFactory.getDefaultedAnnotatedType(tree);
```

Comment:
At the moment, without PR #623 , the default branch is used for all kinds of trees, other than type trees and methods.
To me using getDefaultedAnnotatedTypes here makes sense: take the type the programmer wrote for validation, without any refinements. This makes sense both for the LHS and for any other expression.
Let me try to produce a test case with the PolyAll Checker and compare the the UTS.